packages/polymath-issuer/src/pages/restrictions/components/IndividualRestrictionsTable.js

```diff
@@ -0,0 +1,219 @@
+// @flow
+
+import React, { Component } from 'react';
+import { DataTable, Icon } from 'carbon-components-react';
+import { Button } from '@polymathnetwork/ui';
+import IndividualRestrictionsModal from './IndividualRestrictionsModal';
+import ImportRestrictionsModal from './ImportRestrictionsModal';
+import { connect } from 'react-redux';
+import { modifyIndividualRestriction } from '../../../actions/restrictions';
+import { dispatch } from 'rxjs/internal/observable/range';
+import moment from 'moment';
+const {
+  TableContainer,
+  Table,
+  TableHead,
+  TableRow,
+  TableBody,
+  TableCell,
+  TableHeader,
+  TableToolbar,
+  TableToolbarContent,
+} = DataTable;
+
+const columns = [
+  {
+    header: 'Investor Wallet Address',
+    key: 'address',
+    Cell: ({ value }) => value,
+  },
+  {
+    header: 'Maximum 24h Transfer (# or %)',
+    key: 'dailyRestrictionType',
+    Cell: ({ value }) => value,
+  },
+  {
+    header: 'Start Date',
+    key: 'dailyStartTime',
+    cell: ({ value }) => value,
+  },
+  {
+    header: 'End Date',
+    key: 'dailyEndTime',
+    Cell: ({ value }) => value,
+  },
+  {
+    header: 'Custom Rolling Period Interval',
+    key: 'rollingPeriodInDays',
+    Cell: ({ value }) => value,
+  },
+  {
+    header: 'Maximum Custom Transfer (# or %)',
+    key: 'customRestrictionType',
+    Cell: ({ value }) => value,
+  },
+  {
+    header: 'Start Date',
+    key: 'customStartTime',
+    Cell: ({ value }) => value,
+  },
+  {
+    header: 'End Date',
+    key: 'customEndTime',
+    Cell: ({ value }) => value,
+  },
+];
+
+type State = {|
+  isIndividualRestrictionModalOpen: boolean,
+  isUploadCsvModal: boolean,
+|};
+
+class IndividualRestrictionsTable extends Component<Props, State> {
+  state = {
+    isIndividualRestrictionModalOpen: false,
+    isUploadCsvModal: false,
+  };
+
+  editAddress = address => {
+    const { individualRestrictions } = this.props;
+    let restriction = individualRestrictions.find(i => i.address === address);
+    this.props.modifyIndividualRestriction(restriction);
+    this.setState({ isIndividualRestrictionModalOpen: true });
+  };
+
+  handleCsvOpen = () => {
+    this.setState({ isUploadCsvModal: true });
+  };
+
+  handleCsvClose = () => {
+    this.setState({ isUploadCsvModal: false });
+  };
+
+  handleOpen = () => {
+    this.setState({ isIndividualRestrictionModalOpen: true });
+  };
+
+  handleClose = () => {
+    this.setState({ isIndividualRestrictionModalOpen: false });
+    this.props.modifyIndividualRestriction(null);
+  };
+
+  handleDelete = id => {
+    this.props.removeAddressFromTransferManager(id);
+  };
+
+  formatCell = (id, cell) => {
+    const { individualRestrictions } = this.props;
+    if (cell.value === undefined) return '-';
+    switch (cell.info.header) {
+      case 'customStartTime':
+      case 'dailyStartTime':
+      case 'customEndTime':
+      case 'dailyEndTime':
+        return moment.unix(cell.value / 1000).format('MMM DD YYYY');
```

Comment:
If `cell.value` is brought from VRTM module, it would be in seconds. Perhaps we need to _multiply_ it by `1000` here.

Reply:
The reason it needs to be divided by 1000 cause in redux it is stored with it's * 1000 value. It helps fill the time stamp, otherwise it wouldn't work.